cloud/common.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2024 by Red Hat, Inc. All rights reserved.
+// Use of this source is goverend by the Apache License
+// that can be found in the LICENSE file.
+
+package cloud
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+	"syscall"
+)
+
+// HTTPClient make it easier to swap out the client socket for testing
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// NewClient initializes the values of the cloud API client configuration
+// used to query the server.
+func NewClient(ctx context.Context, socket HTTPClient, socketPath string) Client {
+	// TODO
+	// - check for valid server path
```

Comment:
I'll assume this is about the `socketPath`, what kind of validation do you want to do?

Reply:
Good question. There's a similar comment in the weldr client version of this. I think it may be old. IIRC I was going to check to make sure the socket existed here, but it ends up being easier to use the Exists() and checkSocketError functions since the commands first try cloud and then fall back to weldr if it isn't available. So I'm pretty sure these TODOs can now go away.